### DISCUSSION TITLE
Transfert a .hl7 text file and a .pdf attachment

### DISCUSSION THREAD

**Discussion by labiloute** [nextgenhealthcare/connect#4747] [Q&A] [answered]
Hi all,
I try to create a configuration channel that I thought would be straight forward, but... it's not!

I receive from a sftp serveur a .hl7 text file (containing ORU message) which is joined to a .pdf file.

Both have simply to be transfered to another file system when received.

My issue is encoding. When I use binary, I loose the text file encoding. When I use "text", the PDF file is encoded with the selected encoding, which make it unreadable.

I probably miss something as there should probably exist a simple way to make it !

Sorry about that if it is something dumb.
Regards

**Comment by pacmano1** [comment]
What do you mean by a test filed **joined** to a PDF?  That doesn't make sense.    

Do you mean "I receive 2 files, one a pdf, and one a text file" and they are somehow related to each other?

**Comment by labiloute** [comment]
Hi @pacmano1 ,
That's exactly that. I receive 2 files, that are totally related to each other.
But I don't know how to transfer them from a file system to another without loosing encoding of one of them.
Any idea so far ?
Many thanks

**Reply by tonygermano** [reply]
If you do a binary transfer, there is no loss of encoding. The file will be reproduced exactly as it was read, byte-for-byte. The character encoding is only needed when you are converting between bytes and strings.

**Comment by rjkroll** [comment]
If the servers are on the same network, you could try just using FileUtils.moveFile to send them between shares.

Addtionally, you could try using two destinations, one for the PDF and one for the Text file, with a filter to route the file to whichever destination you need (binary vs text)

**Comment by labiloute** [comment]
Hi @tonygermano & @rjkroll ,
Thanks for your replies.
@tonygermano the sftp server host a .hl7 text file AND a .pdf binary file. If I use the binary transfer I lost the hl7 message. Or maybe I miss something in the config ?
@rjkroll Could you give me an exemple of the FileUtils.moveFile method ? Can I use the source dans destination parameters or I must create the connection in javascript again ?
@rjkroll Would you give me a sample of a two destination filter ? BTW I am worried about the source filetype (text or binary). Which one to choose in this case ?
Thanks for help !

**Reply by tonygermano** [reply]
Like I said, binary will always work for file transfers. The whole reason for there being a binary option is when you need to read and/or write a file without potentially changing it. The contents of the file do not matter. Can you share your channel where you tried to do a binary transfer and lost a file? Did you use binary in both the file reader and the file writer?

**Reply by labiloute** [reply]
Next time I'll try this out !

**Comment by labiloute** [comment, marked as answer]
Hi @tonygermano .
Thanks for your help.
I forgot to say that I may have to make modifications on the text file.
So far, I only had to change encoding of .hl7 file from UTF8 to ISO-8859-1, but I obviously don't want to apply on the attached pdf file.
So I keep your advince for a future channel, thank you.

I managed to make it with two different channels :
 - first full binary and full "delimited text" in "Summary" > "Set Data types"
 - second one in UTF8 inbound and ISO8859-1 outbound
 - both on the same source and same destination

Now, finger crossed, I have to hope that files will arrive at the same time on the destination or it will generate an error on destination software.

Anyway I thank you and I will close this thread tomorrow if everything works as expected

Note : check the solution provided above by @tonygermano first !
